custom_model_runner/datarobot_drum/resource/components/Python/generic_predictor/generic_predictor.py

```diff
@@ -53,11 +58,43 @@ def configure(self, params):
 
     def _materialize(self, parent_data_objs, user_data):
         input_filename = self._params["input_filename"]
-        predictions = self._predictor.predict(input_filename)
         output_filename = self._params.get("output_filename")
+
+        additional_params = self._params.get("query_params")
+
         if self._target_type == TargetType.UNSTRUCTURED:
-            with open(output_filename, "w") as f:
-                f.write(predictions)
+            kwargs_params = {}
+
+            # trick in order not to write own parser for params using
+            # "name1=value1;name2=value2" schema
+            _, params = werkzeug.http.parse_options_header("_;" + additional_params)
```

Comment:
we should not be using header parser for query.

either param per query value or a real query parser, consistent with web browser.

```
[ins] In [4]: from urllib.parse import parse_qs
[ins] In [5]: parse_qs("search=quint&tags=python")
Out[5]: {'search': ['quint'], 'tags': ['python']}
```

Reply:
switched to
```
>>> dict(parse_qsl("name1=val1;name2=[val2,val3]"))
{'name1': 'val1', 'name2': '[val2,val3]'}
```